remoteappmanager/docker/container_manager.py

```diff
@@ -165,12 +165,24 @@ def containers_from_filters(self, filters):
         containers = []
         infos = yield self.docker_client.containers(filters=filters)
         for info in infos:
-            container = Container.from_docker_containers_dict(info)
+            try:
+                container = Container.from_docker_containers_dict(info)
+            except Exception:
+                self.log.exception("Unable to parse container info.")
+                continue
 
             # override the ip and port obtained by the docker info with the
             # appropriate ip and port, considering that we might be using a
             # separate docker machine
-            ip, port = yield from self._get_ip_and_port(container.docker_id)
+            try:
+                ip, port = yield from self._get_ip_and_port(
+                    container.docker_id)
+            except RuntimeError:
+                self.log.exception(
+                    "Unable to retrieve ip/port "
+                    "for container {}".format(container.docker_id))
```

Comment:
Should it append to the containers here anyway?  This method looks for containers with a given set of filters, it would be a surprise if the results are further filtered by the existence of ip/port.